src/orchestration/Orchestration.ts

```diff
@@ -213,6 +233,13 @@ export class Orchestration {
 
         this.config.endpoint = this.getDataPlaneEndpoint(region, partialConfig);
 
+        // If the URL is not formatted correctly, a TypeError will be thrown.
```

Comment:
question: Would it be worth adding a unit test case to verify URLs that is expected to fail do fail? (even though URL is not our code).